### PR TITLE
Add documentation for optional questions

### PR DESCRIPTION
**Pull request by plannigan**
* Provide explanations and examples for optional questions and branching paths.
* Create an examples directory for the code examples used in the documentation.

This addition starts to make the usage page large. With more to come, this will be hard to manage. A follow up PR will break this page down into smaller more focused pages.

### CODE REVIEW

File: docs/usage-guide.md [hunk 1]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
 `Confirm` doesn't take any additional arguments that weren't [mentioned above](#all-questions). However, the `default`
 argument takes a `bool` instead of `str`.
 
+## Only Asking Some Questions
+
+There are situations where a question should be asked some times, but not all the time. A program that is collects
+information about a user's pets should not ask the user for the dog's name and breed if the user said they do not have a
```
Comment (jdw6359) [inline docs/usage-guide.md:79]:
```suggestion
There are situations where a question should be asked some times, but not all the time. For example, a program that collects
information about a user's pets should not ask the user for the dog's name and breed if the user said they do not have a
```

File: docs/usage-guide.md [hunk 2]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
...
+        should_ask=does_user_have_a_dog,
+        default="Basset Hound"
+    )
+]
+
+user_answers = columbo.get_answers(interactions)
+print(user_answers)
+```
+
+If the user accepts the default answers for each of these questions, the output will be.
```
Comment (jdw6359) [inline docs/usage-guide.md:121]:
```suggestion
If the user accepts the default answers for each of these questions, the output will be:
```

File: docs/usage-guide.md [hunk 3]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
...
+
+```
+{"has_dog": True, "dog_name": "Kaylee", "dog_breed": "Basset Hound"}
+```
+
+However, when the user answers the first question with no, the output will be.
```
Comment (jdw6359) [inline docs/usage-guide.md:127]:
```suggestion
However, when the user answers the first question with "no", the output will be:
```

File: docs/usage-guide.md [hunk 4]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
...
+
+```
+{"has_dog": False}
+```
+
+Note that in this case, the `Answers` dictionary only has an answer to the first question.
+
+### Branching Paths
+
+A question that is part of a branching path very similar to an optional question. It is still a question where
```
Comment (jdw6359) [inline docs/usage-guide.md:137]:
```suggestion
A question that is part of a branching path is very similar to an optional question. It is still a question where
```

File: docs/usage-guide.md [hunk 5]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
...
+`should_ask` was given to determine if the question should be skipped our not. The branching aspect comes there being at
+least two sets of optional questions. Each set has a `should_ask` argument that checks for a different state for a
```
Comment (jdw6359) [inline docs/usage-guide.md:139]:
```suggestion
the `should_ask` function was provided to determine if the question should be skipped or not. The branching aspect comes there being at
least two sets of optional questions. Each set has a `should_ask` argument that checks for a different state for a
```

File: docs/usage-guide.md [hunk 6]
```diff
@@ -74,6 +73,140 @@ In addition to the arguments [mentioned above](#all-questions), `Choice` also ac
...
+single answer. In this way, only one of the sets of optional questions will ever be asked.
+
+The following is an example of a short story that has two divergent paths that join at the end. Each individual question
+isn't different from the optional questions [demonstrated above](#optional-questions). The program achieves the
```
Comment (jdw6359) [inline docs/usage-guide.md:143]:
```suggestion
The following is an example of a short story that has two divergent paths that join at the end. Each individual question
is an example of the optional questions [demonstrated above](#optional-questions). The program achieves the
```

File: examples/README.md [hunk 7]
```diff
@@ -0,0 +1,6 @@
+# Examples
+
+A collection of examples that can be run to see how `columbo` works. If it not clear what a specific example is doing,
+please read the [documentation][documentation]. These examples all appear there with detailed explications.
```
Comment (jdw6359) [inline examples/README.md:4]:
```suggestion
please read the [documentation][documentation]. These examples all appear there with detailed explanations.
```

**Review by jdw6359 — APPROVED** [review]

File: docs/usage-guide.md (hunk 6 above)
Comment (plannigan) [reply docs/usage-guide.md:143]: What I was trying to convey here is that questions used for branching are optional questions used in a specific way. I think the suggested change moves the sentence away from conveying that information. Do you have an idea that would better describe that aspect of the concept?

**Pull request merged**